### PR TITLE
update message manager

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by krzysztofwolski** [comment]
@AlicjaSzu when testing new messages with storybook, message timer is not stopping when mouse is on message. Will it be added in separate PR?

File: src/index.tsx [hunk 1]
```diff
@@ -1,65 +1,66 @@
...
-import ConfigurationSection, { createConfigurationMenu } from "./configuration";
-import AppStateProvider from "./containers/AppState";
-import BackgroundTasksProvider from "./containers/BackgroundTasks";
 import { CustomerSection } from "./customers";
+import { DateProvider } from "./components/Date";
 import DiscountSection from "./discounts";
+import ErrorBoundary from "react-error-boundary";
 import HomePage from "./home";
-import { commonMessages } from "./intl";
+import { LocaleProvider } from "./components/Locale";
```
Comment (dominik-zeglen) [inline src/index.tsx:26]: Something wrong happened here with import sorting 🙊

**Comment by AlicjaSzu** [comment]
Task linked: [CU-60rppx Handle timer for closing notifications](https://app.clickup.com/t/60rppx)

**Comment by AlicjaSzu** [comment]
Task linked: [CU-60rppa Implement info and warning notification levels](https://app.clickup.com/t/60rppa)

**Comment by AlicjaSzu** [comment]
Task linked: [CU-60rpqx Add ability to expand description](https://app.clickup.com/t/60rpqx)

**Comment by AlicjaSzu** [comment]
Task linked: [CU-60rtv2 Handle multiple notifications on screen](https://app.clickup.com/t/60rtv2)

**Comment by krzysztofwolski** [comment]
@AlicjaSzu error message is closing itself at storybook. DoD specified that error messages will not have timers (https://app.clickup.com/t/60rppx)

File: src/components/messages/MessageManagerProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  const remove = useCallback(notificationId => {
+    setNotifications(currentNotifications =>
+      currentNotifications.filter(n => n.id !== notificationId)
+    );
+  }, []);
+
+  const show = useCallback((message = {}, timeout = 3000) => {
+    const id = Math.random()
+      .toString(36)
+      .substr(2, 9);
```
[the remark this answers is not in the document]
Comment (AlicjaSzu) [reply src/components/messages/MessageManagerProvider.tsx:56]: i was thinking about it

**Pull request merged**